docs/docs/build-caching.md

```diff
@@ -3,7 +3,70 @@ title: Build Caching
 issue: https://github.com/gatsbyjs/gatsby/issues/8103
 ---
 
-This is a stub. Help our community expand it.
+Plugins can cache data as json objects and retrieve them on consecutive builds.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+Caching is already used by Gatsby and plugins for example:
+
+* any nodes created by source/transformer plugins are cached
+* `gatsby-plugin-sharp` caches built thumbnails
+
+
+Data is stored in the `.cache` directory relative to your project root.
+
+
+## The cache api
+
+The cache API is passed to [Gatsby's Node APIs](/docs/node-apis/), which is typically implemented by plugins.
+```
+exports.onPostBootstrap = async function({ cache, store, graphql }) {
+```
+
+
+The two functions you would want to use are:
+
+### `set`
```

Comment:
Can you describe what these two functions do for those who are new to plugins, particularly when/why you would use them?